nixos/modules/services/web-apps/collabora-online.nix

```
@@ -0,0 +1,1638 @@
+
+          <help_url desc="The Help root URL, or empty for no help (hides the Help buttons)" type="string" default="https://help.collaboraoffice.com/help.html">${toStr c.help_url}</help_url>
+
+          <overwrite_mode>
+              <enable desc="Enable overwrite mode (user can use insert key)" type="bool" default="true">${toStr c.overwrite_mode.enable}</enable>
+          </overwrite_mode>
+
+          <wasm desc="WASM-specific settings">
+              <enable desc="Enable WASM support" type="bool" default="false">${toStr c.wasm.enable}</enable>
+              <force desc="When enabled, all requests are redirected to WASM." type="bool" default="false">${toStr c.wasm.force}</force>
+          </wasm>
+
+      </config>
+    '';
+
+  upstreamdoc = ''
+    See [upstream example](https://github.com/CollaboraOnline/online/blob/master/coolwsd.xml.in) for the reference.
+  '';
+
+  logging_level_type = lib.types.oneOf [
+    (lib.types.ints.between 0 8)
+    (lib.types.enum [
+      "none" # 0
+      "fatal" # 1
+      "critical" # 2
+      "error" # 3
+      "warning" # 4
+      "notice" # 5
+      "information" # 6
+      "debug" # 7
+      "trace" # 8
+    ])
+  ];
+in
+{
+  options.services.collabora-online = {
+    enable = lib.mkEnableOption "collabora-online";
+
+    package = lib.mkPackageOption pkgs "collabora-online" { };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 9980;
+      description = "Listening port";
+    };
+
+    config = {
+      # Accessibility settings
+      accessibility = {
+        enable = lib.mkEnableOption "accessibility support";
+      };
+
+      allowed_languages = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        default = [
+          "de_DE"
+          "en_GB"
+          "en_US"
+          "es_ES"
+          "fr_FR"
+          "it"
+          "nl"
+          "pt_BR"
+          "pt_PT"
+          "ru"
+        ];
+        description = ''List of supported languages of Writing Aids (spell checker, grammar checker, thesaurus, hyphenation) on this instance. Allowing too many has negative effect on startup performance.'';
+      };
+
+      # Remote API settings for spell and grammar checking.
+      # These are the settings of external (remote) spellchecker and grammar checker services. Currently LanguageTool and Duden Korrekturserver APIs are supported, you can
+      # set either of them. By default they are disabled. To turn the support on, please set "enabled" property to true. It works with self hosted or cloud services, free
+      # and premium as well. The "base_url" may be https://api.languagetoolplus.com/v2 if the cloud version of LanguageTool is used. Please note that your data in the
+      # document e.g. the text part of it will be sent to the cloud API. Please read the respective privacy policies, e.g. https://languagetool.org/legal/privacy.
+      languagetool = {
+        enabled = lib.mkEnableOption ''Remote Spell and Grammar Checker'';
```

```suggestion
        enabled = lib.mkEnableOption "Remote Spell and Grammar Checker";
```
Please use single quotes for all similar short mkEnableOption and description